airflow/models/asset.py

```diff
@@ -37,8 +39,30 @@
 from airflow.models.base import Base, StringID
 from airflow.settings import json
 from airflow.utils import timezone
+from airflow.utils.session import NEW_SESSION, provide_session
 from airflow.utils.sqlalchemy import UtcDateTime
 
+if TYPE_CHECKING:
+    from typing import Sequence
+
+    from sqlalchemy.orm import Session
+
+
+@provide_session
```

Comment:
We should not use `provide_session` here. The caller should always provide a session instead.

Reply:
Sure. Will remove it.

Reply:
removed